router/src/panic.c

```diff
@@ -0,0 +1,332 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+//
+// Code for unwinding the stack and printing debug information when a non-recoverable signal occurs.
+//
+
+#define _GNU_SOURCE  // to get gettid()
+
+#include "config.h"
+
+#include "qpid/dispatch/threading.h"
+
+#include <assert.h>
+#include <errno.h>
+#include <signal.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#ifdef HAVE_LIBUNWIND
+#define UNW_LOCAL_ONLY
+#include <libunwind.h>
+#endif
+
+static void panic_signal_handler(int signum, siginfo_t *siginfo, void *ucontext);
+
+// Define those signals which will be handled.
+//
+typedef struct {
+    int         signal;
+    const char *name;
+} panic_signal_info_t;
+
+// clang-format off
+static const panic_signal_info_t panic_signals[] = {
+    { .signal = SIGABRT, .name = "SIGABRT" },
+    { .signal = SIGBUS,  .name = "SIGBUS" },
+    { .signal = SIGFPE,  .name = "SIGFPE" },
+    { .signal = SIGILL,  .name = "SIGILL" },
+    { .signal = SIGSEGV, .name="SIGSEGV" },
+    {0}
+};
+// clang-format on
+
+/**
+ * Install the panic signal handler. This is done early in the router initialization so do not attempt to log or use the
+ * alloc pool, etc. This is not called during a signal handler so there is no need to avoid async signal unsafe calls.
+ */
+void panic_handler_init(void)
+{
+    if (getenv("SKUPPER_ROUTER_DISABLE_PANIC_HANDLER") == 0) {
+        struct sigaction sa = {
+            // use SA_RESETHAND since if the stack unwind fails the default signal handler (coredump) will be invoked
```

Comment:
the comment is incorrect, no? the default signal handler will be invoked at the end _regardless of whether stack unwind failed or not_. Which is the desired behavior, so that when we can get coredump, we do also get coredump, in addition to stack unwind.

Reply:
Good catch - remove.